packages/legend-graph/src/__tests__/ImportResolutionRoundtrip.test.ts

```diff
@@ -205,3 +213,20 @@ describe(unitTest('Service import resolution roundtrip'), () => {
     },
   );
 });
+
+test(unitTest('Versioned class'), async () => {
+  const graphManagerState = TEST__getTestGraphManagerState();
+  const data = TEST_DATA__VersionedClassRoundtrip as Entity[];
+  await TEST__buildGraphWithEntities(graphManagerState, data, {
+    TEMPORARY__keepSectionIndex: true,
+  });
+
+  const transformedEntities = graphManagerState.graph.allOwnElements.map(
```

Comment:
There is no need to convert the graph to entities. Once converted they become type `Entity` thus your if statement below will never be reached. 

We can asssert directly on the classes in the graph.`graphManagerState.graphManager.graph`